### ISSUE TITLE
Fail to load plugin

### ISSUE THREAD

**Issue by soez** [PaoloMonti42/salt#1]
Hi, I get the following error when loading gdb and I can not use the plugin

> Traceback (most recent call last):
>   File "/home/noname/salt/salt.py", line 6, in <module>
>     cpu0_offset = gdb.lookup_global_symbol('__per_cpu_offset').value()[0]
> AttributeError: 'NoneType' object has no attribute 'value'

regards

**Comment by PaoloMonti42** [comment]
Hi! Glad to see you gave this tool a try :)
What kernel version are you trying to work on? What symbol file are you loading on gdb?
You should verify that your file contains the "__per_cpu_offset" symbol through the _nm_ command.

**Comment by soez** [comment]
The file I am opening is an exploit of a ctf challenge. Regarding if it has the symbol "__per_cpu_offset" it does not have it. The kernel version is shown below.

> noname@ubuntu$ nm exp | grep __per_cpu_offset
> noname@ubuntu$ uname -r
> 4.18.0-22-generic

**Comment by PaoloMonti42** [comment]
This plugin in meant to be used when gdb is attached to the kernel itself, not an userspace binary :) then, from inside the debugged kernel, you should run the exploit
In fact the symbol file I'm referring to is the kernel symbols, you should download the correct one for your version

**Comment by soez** [comment]
true!! I had not fallen into the situation, I should have debugged vmlinux... thanks!
